gobblin-data-management/src/test/resources/manifestBasedDistcpTest/sampleManifest.json

```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "id":"1",
+    "fileName":"/tmp/dataset/test1.txt",
+    "fileGroup":"/tmp/dataset",
+    "fileSizeInBytes":"1024"
+  },
```

Comment:
I'm glad you created this, so we have something to verify the parsing against, but I don't believe it is actually exercised in the included unit test.  that only constructs the `ManifestBasedDataset`, but I believe we must call `.getFilesetIterator` for parsing to kick in, no?  if that's true, let's add such a test